internal/host/host_test.go

```diff
@@ -2651,3 +2652,59 @@ var _ = Describe("Disabled Host Validation", func() {
 	})
 
 })
+
+var _ = Describe("Get host by Kube key", func() {
+	var (
+		state            API
+		ctrl             *gomock.Controller
+		db               *gorm.DB
+		key              types.NamespacedName
+		kubeKeyNamespace = "test-kube-namespace"
+		dbName           string
+		mockEvents       *events.MockHandler
+		mockHwValidator  *hardware.MockValidator
+		validatorCfg     *hardware.ValidatorCfg
+		id               strfmt.UUID
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		ctrl = gomock.NewController(GinkgoT())
+		db, dbName = common.PrepareTestDB()
+		mockEvents = events.NewMockHandler(ctrl)
+		mockHwValidator = hardware.NewMockValidator(ctrl)
+		validatorCfg = createValidatorCfg()
+		mockMetric := metrics.NewMockAPI(ctrl)
+		db, dbName = common.PrepareTestDB()
+		state = NewManager(common.GetTestLog(), db, mockEvents, mockHwValidator, nil, validatorCfg, mockMetric, defaultConfig, nil, nil)
+		id = strfmt.UUID(uuid.New().String())
+		key = types.NamespacedName{
+			Namespace: kubeKeyNamespace,
+			Name:      string(id),
```

Comment:
no need for the casting.
```suggestion
			Name:      id.String(),
```